source/core/list.h

```diff
@@ -504,7 +504,7 @@ namespace Slang
 		template<typename T2>
 		UInt IndexOf(const T2 & val) const
 		{
-			for (int i = 0; i < _count; i++)
```

Comment:
Yeah - I think using a signed type for an index type, overall is the better choice. The main advantage of unsigned types here is you can't pass in a negative index, but in practice it causes more problems. Against it - the code can be faster with say int (compiler has more opportunities to optimize, because less constraints, and int is faster on some archs, because more common), there is an obvious 'not found' index and indexing maths and simpler index maths (like counting downwards).

Reply:
The funny thing is that I've heard different arguments for why pointer-sized ints are the right default, because they don't need to be sign-extended when you add them to pointers (for example).

I guess the point is that by using `Int` and `UInt` as our general-purpose types we can potentially typedef them to whatever turns out to be fastest in practice.

Reply:
It's nice to have that flexibility. I have used a type specific to indexing for this specific purpose in the past 'IndexT'. I also in the past had UInt and Int, but they were mode about consistency across the type system and to make switchable if needed.